ibm_cloud_sdk_core/authenticators/__init__.py

```diff
@@ -13,10 +13,82 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""The ibm_cloud_sdk_core project supports the following types of authentication:
+
+  Basic Authentication
+  Bearer Token
+  Identity and Access Management (IAM)
+  Cloud Pak for Data
+  No Authentication
+
+  The authentication types that are appropriate for a particular service may vary from service to service.
+  Each authentication type is implemented as an Authenticator for consumption by a service.
+
+classes:
+  Authenticator: Abstract Base Class. Implement this interface to provide custom authentication schemes to services.
+  BasicAuthenticator: Authenticator for passing supplied basic authentication information to service endpoint.
+  BearerTokenAuthenticator: Authenticator for passing supplied bearer token to service endpoint.
+  CloudPakForDataAuthenticator: Authenticator for passing CP4D authentication information to service endpoint.
+  IAMAuthenticator: Authenticator for passing IAM authentication information to service endpoint.
+  NoAuthAuthenticator: Performs no authentication. Useful for testing purposes.
+
+functions:
+  get_authenticator_from_environment: Get authenticator from external sources.
+"""
 
 from .authenticator import Authenticator
 from .basic_authenticator import BasicAuthenticator
 from .bearer_token_authenticator import BearerTokenAuthenticator
 from .cp4d_authenticator import CloudPakForDataAuthenticator
 from .iam_authenticator import IAMAuthenticator
 from .no_auth_authenticator import NoAuthAuthenticator
+from ..utils import read_external_sources
+
+def get_authenticator_from_environment(service_name: str) -> Authenticator:
```

Comment:
As we discussed, I think we should try to expose the `get_authenticator_from_environment` function in a way that doesn't create an incompatibility with existing users of the python core v1.0.0.

Reply:
this is not the latest and has been fixed.